src/Chapter2.hs

```diff
@@ -842,7 +857,9 @@ list.
 
 🕯 HINT: Use the 'cycle' function
 -}
-rotate = error "rotate: Not implemented!"
+rotate :: Int -> [a] -> [a]
+rotate x _ | x < 0 = []
+rotate x xs = take (length xs) $ drop x $ cycle xs
```

Comment:
You could have a bit more efficient solution if you would check on an empty list and if `x` is zero (by module of the length as well) 🙂